src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -407,14 +407,15 @@ protected function denormalizeParameter(\ReflectionClass $class, \ReflectionPara
     }
 
     /**
-     * @param array  $parentContext
-     * @param string $attribute
+     * @param array       $parentContext
+     * @param string      $attribute     Attribute name
+     * @param string|null $format
      *
      * @return array
      *
      * @internal
      */
-    protected function createChildContext(array $parentContext, $attribute)
+    protected function createChildContext(array $parentContext, $attribute/*, ?string $format*/)
```

Comment:
should i also trigger a deprecation warning here? we don't actually use the format at all...

Reply:
not on 3.4 but on master yes - also, this should be `/*, string $format = null */`, isn't it?

Reply:
@dbu Can you fix the typos? (no `?` and a space before `*/`?)

Reply:
as its a protected function, i actually think ?string would make more sense, you will always pass the parameter even when its null. nullable types is php 7.2 only, which i guess symfony 5 will require at least. i changed it to the optional argument syntax for now.

Reply:
also, if its indeed an optional argument, the deprecation warning would be unnecessary.

Reply:
ok, fair enough

Reply:
shall i change it to nullable argument? the nullable syntax was introduced in 7.1 actually. or shall i only change it on master when i also add a deprecation warning?